StripeCore/StripeCore/Source/Attestation/StripeAttest.swift

```diff
@@ -117,33 +117,42 @@ import UIKit
         case successfullyAttested = "STPAttestKeySuccessfullyAttested"
     }
 
-    private static var keyID: String? {
+    private var keyID: String? {
         get {
-            UserDefaults.standard.string(forKey: DefaultsKeys.keyID.rawValue)
+            UserDefaults.standard.string(forKey: defaultsKeyForSetting(.keyID))
         }
         set {
-            UserDefaults.standard.set(newValue, forKey: DefaultsKeys.keyID.rawValue)
+            UserDefaults.standard.set(newValue, forKey: defaultsKeyForSetting(.keyID))
         }
     }
 
-    private static var successfullyAttested: Bool {
+    private var successfullyAttested: Bool {
         get {
-            UserDefaults.standard.bool(forKey: DefaultsKeys.successfullyAttested.rawValue)
+            UserDefaults.standard.bool(forKey: defaultsKeyForSetting(.successfullyAttested))
         }
         set {
-            UserDefaults.standard.set(newValue, forKey: DefaultsKeys.successfullyAttested.rawValue)
+            UserDefaults.standard.set(newValue, forKey: defaultsKeyForSetting(.successfullyAttested))
         }
     }
 
-    private static var lastAttestedDate: Date? {
+    private var lastAttestedDate: Date? {
         get {
-            UserDefaults.standard.object(forKey: DefaultsKeys.lastAttestedDate.rawValue) as? Date
+            UserDefaults.standard.object(forKey: defaultsKeyForSetting(.lastAttestedDate)) as? Date
         }
         set {
-            UserDefaults.standard.set(newValue, forKey: DefaultsKeys.lastAttestedDate.rawValue)
+            UserDefaults.standard.set(newValue, forKey: defaultsKeyForSetting(.lastAttestedDate))
         }
     }
 
+    /// The key to use for storing an attestation key in NSUserDefaults.
+    func defaultsKeyForSetting(_ setting: SettingsKeys) -> String {
+        var key = "\(setting.rawValue):\(apiClient.publishableKey ?? "unknown")"
```

Comment:
Storing "unknown:xxx" in NSUserDefaults feels awkward. If we don't have a publishable key, seems like we should just bail?

Reply:
Let's bail above in `getOrCreateKey()` (though I'll still safely unwrap here so we don't crash)

Reply:
Maybe we take a copy of apiClient.publishableKey when the class is initialized?